numpy/random/generator.pyx

```diff
@@ -3829,6 +3834,8 @@ cdef class Generator:
             # Fast, statically typed path: shuffle the underlying buffer.
             # Only for non-empty, 1d objects of class ndarray (subclasses such
             # as MaskedArrays may not support this approach).
+            if axis != 0 and axis != -1:
+                raise np.AxisError(axis, x.ndim)
```

Comment:
For simplicity, can you just call `normalize_axis_index ` unconditionally at the start of the function?

Reply:
@eric-wieser sure.

Reply:
Note that you'll need to use `np.ndim(x)` instead of `x.ndim`